src/main/java/com/google/sps/utils/GameUtils.java

```diff
@@ -123,4 +124,59 @@ public static boolean addUserToGame(String userId, Entity gameEntity, DatastoreS
     
     return true;
   }
-}
\ No newline at end of file
+ 
+  public static String joinGame(Entity userEntity, String gameName, DatastoreService datastore, String gameId) {
```

Comment:
This pattern -- returning a String to indicate the outcome of a function call -- is error-prone and makes it difficult to use `joinGame()` properly. It's better practice to assume the caller will test for null, and throw `NullPointerExeption` if they do not.

As for the case where the user is already in the game, there are two ways to handle it:
1. Throw a custom `Exception` subclass, called e.g. `UserAlreadyInGameException`.
2. Don't tell the caller whether you added the user, or whether the user was already in the game. Does the caller really need to know?

I prefer option 2, because it makes life simpler for the caller. If they really needed to know whether the user was part of the game already, they could have called `IsValidGameName`.

Reply:
+1

Reply:
Alright, I changed the function and moved some of the checks around; some of it's back in the servlet, and the function only returns true or false. It can still be called by both start and join, because it is called after the game is created.